core/MatomoInstaller.php

```diff
@@ -0,0 +1,416 @@
+        self::tablesCreation();
+        self::setupSuperUser($settings);
+        self::firstWebsiteSetup($settings);
+    }
+
+    /*
+     * Check if an headless installation is expected.
+     * (Heritance should be related to it)
+     */
+    public static function isHeadlessInstall()
+    {
+        $installSettings = Config::getInstance()->MatomoInstall;
+        return !empty($installSettings);
+    }
+
+    /*
+     * Use settings from MatomoInstall section in
+     * the config.ini.php file and perform installation.
+     */
+    public static function installFromConfig()
+    {
+        //Use settings defined in config.ini.php
+        //in MatomoInstall section.
+        $settings = Config::getInstance()->MatomoInstall;
+        self::headlessInstall($settings);
+
+        //Remove installation config when finished
+        Config::getInstance()->MatomoInstall = array();
+        Config::getInstance()->forceSave();
+
+        //Redirect to matomo homepage
+        Url::redirectToUrl('index.php');
+    }
+
+    /**
+     * Installation Step 2: System Check
+     *
+     * Control if the system is ready for installation,
+     * run Plugins\Diagnostics to get state.
+     * 
+     * @view View (optionnal), view to store DiagnosticReport
+    */
+    //S: Should I perform the system check ?
+    //In : https://github.com/digitalist-se/extratools/blob/master/Lib/Install.php
+    // no system check seem to be performed.
+    public static function systemCheck($view=null)
+    {
+        //Why could delete conf be useful ?? Should I handle
+        //already created config ? + delete conf on failure.
+        // ! Having an error if I remove it.
+        // Multiple entrence for installation, must know where plug
+        // the install.
+        self::deleteConfigFileIfNeeded();
+        // Do not use dependency injection because this service requires a lot of sub-services across plugins
+        /** @var DiagnosticService $diagnosticService */
+        $diagnosticService = StaticContainer::get('Piwik\Plugins\Diagnostics\DiagnosticService');
+        $diagnosticReport = $diagnosticService->runDiagnostics();
+
+        //Store the diagnostic report in the view
+        // !! Should remove Installation plugin mecanism
+        //Must think about the way to handle error here
+        if (!is_null($view))
+        {
+            $view->diagnosticReport = $diagnosticReport;
+            $view->showNextStep = !$diagnosticReport->hasErrors();
+
+            // On the system check page, if all is green, display Next link at the top
+            $view->showNextStepAtTop = $view->showNextStep && !$diagnosticReport->hasWarnings();
+        }
+        //HEADLESS: DO STUFF WITH errors
+        return (!$diagnosticReport->hasErrors());
+    }
+
+    /**
+     * Installation Step 3: Database Set-up
+     * @throws Exception|Zend_Db_Adapter_Exception
+     */
+    public static function createDatabaseObject($settings)
+    {
```

Comment:
Wondering if it would be better to directly configure the database in the database section instead of the new `Installer` section. The Installer could simply check if the database is configured correctly and if no tables exist yet, the installer could create them...